# **Basic Python**

This notebook introduces the basic syntax of Python

\
**Instruction: type your code when you see 'Type your code here' in comments**

# exercise 1

Write a program to ask the user for his/her name


In [ ]:
#Type your code here
name = input('what\'s your name?')
print('hello %s!' % name)


hello dmdd!


# exercise2

2.1 Try to run the codes below, what do you see?

In [ ]:
b = 12345 
a = str(b) 
print(type(a))
a

<class 'str'>


'12345'

2.2 Try to convert the followings to bool(), what do you see?




1.   '' (This presents empty string)
2.   'abc'
3.   1
4.   0
5.   '0'
6.   '#%@#$%'





2.2.1 

In [ ]:
#Type your code here
bool('')
#False

False

2.2.2

In [ ]:
#Type your code here
bool('abc')
#True

True

2.2.3


In [ ]:
#Type your code here
bool(1)
#True

True

2.2.4

In [ ]:
#Type your code here
bool(0)
#False

False

2.2.5

In [ ]:
#Type your code here
bool('0')
#True

True

2.2.6

In [ ]:
#Type your code here
bool('#%@#$%')
#True

True

# exercise3


a=123.45

b='456.78'

a+b

In [ ]:
#Type your code here

a = 123.45
b = '456.78'
a + b

TypeError: unsupported operand type(s) for +: 'float' and 'str'

Interpret the variable **b** as a floating point number and add it to the variable **a**

In [ ]:
#Type your code here
a = 123.45
b = '456.78'

print(a + float(b))

580.23


Interpret the variable **a** as a string and concatenate it with the variable **b**

In [ ]:
#Type your code here
a = 123.45
b = '456.78'

print(str(a) + b)

123.45456.78


# exercise4

There are 3 people: John, Matt and Brian, here are their weights (in kg):



<!DOCTYPE html>
<html>
<head>
<style>

table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
  font-size: 10px;
  text-align:  center;
}

td,th {
  border: 10px solid #dddddd;
  text-align:  center;
  font-size: 20px;
  padding: 8px;
}

tr:nth-child(even) {
  background-color: #dddddd;
}
</style>
</head>
<body>


<table>
  <tr>
    <th>Name</th>
    <th>Weight (in kg)</th>
  </tr>

  <tr>
    <td>John</td>
    <td>60</td>
  </tr>

  <tr>
    <td>Matt</td>
    <td>70</td>
  </tr>

  <tr>
    <td>Brian</td>
    <td>90</td>
  </tr>

</table>

</body>
</html>

Write a program to calculate the sum and the mean of their weights,
then print them out to the console.

In [ ]:
#Type your code here

weight = [60, 70, 90]
print('sum of weight = %d' % sum(weight))
print('mean of weight  = %d' % (sum(weight)/len(weight)))


sum of weight = 220
mean of weight  = 73


# exercise 5



*   Declare two variables **x** and **y** with the same value 0.1
*   See what the “==” and “is” operator return
*   Check the ids of the variables
*   Explain the result







In [ ]:
#Type your code here
x = y = 0.1
x == y
#True

True

Check the ids of the variables

In [ ]:
#Type your code here
x = y = 0.1
x is y
#True

True

# exercise 6

* Try to create a “tuple” with one element like this:

>  number  = (1)




* Check the type with the function type(), what do you see?

* What about a tuple with a comma?

>  number  = (1,)

* Explain your result


In [ ]:
#Type your code here
number = (1)
number0 = (1,)
print(type(number))
print(type(number0))
#<class 'int'>
#<class 'tuple'>

<class 'int'>
<class 'tuple'>


# exercise 7

* Create a dictionary
> student = {'name':'Mary', 'name':'Clarissa', 'age':22, 'major':'business'}
* Get the name with student[‘name’], what do you see?
* Print out the dictionary, what do you see?
* Explain the result



In [ ]:
#Type your code here
student = {'name':'Mary', 'name':'Clarissa', 'age':22, 'major':'business'}
student['name']
student.get('name')
#Clarissa
#There are two same key name representing different value.
#the computer can only recognise one of them.

'Clarissa'

# exercise 8

* Write a program to ask the user to input a valid phone number
* Print out the input phone number and tell whether it is a valid phone number
* A valid phone number always:

>1. Has 8 digits

>2. Start with the number 2, 3, 5, 6, or 9


Tips: the input() function always returns a string value



In [ ]:
## Type your code here

number = input('input a valid phone number:')
try:
    int(number)
except ValueError: 
    print('invalid input: %d is not a phone number' % number)
    continue
if len(number) != 8 or int(number[0]) not in [2,3,5,6,9]:
    print('invalid input: %d is not a phone number' % number)
else:
    print('this is a valid phone number!')


input a valid phone number:test
invalid input
input a valid phone number:1234
invalid input
input a valid phone number:12345678
invalid input
input a valid phone number:66666666
this is a valid phone number!


# exercise 9

* Write a program to print a sequence of odd number 
* Starting from 11
* Until 25 (inclusive)

Tips: you can specify the ending of the print() function with print(item, end=’ ‘)



In [ ]:
#Type your code here
print(list(filter(lambda x : x % 2 , range(11,26))))
print(list(range(11,26,2)))

[11, 13, 15, 17, 19, 21, 23, 25]
[11, 13, 15, 17, 19, 21, 23, 25]


# exercise10

* Back to the phone number exercise
* Can you modify it such that it keeps asking until the number is a valid phone number?
* Does your program behave correctly in all test cases?

* Test cases:
> 1. 9876543 (not valid)
> 2. 2345 (not valid)
> 3. 12345678 (not valid)
> 4. 23456789 (valid)
> 5. 65218819 (valid)




In [ ]:
#Type your code here
while True:
    number = input('input a valid phone number:')
    try:
        int(number)
    except ValueError: 
        print('invalid input: %s is not a phone number' % number)
        continue
    if len(number) != 8 or int(number[0]) not in [2,3,5,6,9]:
        print('invalid input: %s is not a phone number' % number)
    else:
        break
print('this is a valid phone number!')

input a valid phone number:9876543
invalid input: 9876543 is not a phone number
